Add tests for CategoriesController

VaolEr opened the pull request.

VaolEr committed:
Add tests for CategoriesController;

VaolEr committed:
Fixed update test in UserCategoriesControllerTests;

dimio commented on src/test/java/com/example/storehouse/web/categories/AdminCategoriesControllerTests.java:
```diff
@@ -0,0 +1,178 @@
+        verify(categoriesService).delete(absentedCategoryId);
+    }
+
+    //TODO вынести в файл TestData?
```
Если в TestData - то тогда уже логично будет его перекрестить в TestUtil
Либо вынести в абстрактный, который выше. Тем более - TestData у нас и для сервисов используется, а авторизация нужна только в контроллерах. Думаю - лучше поднять выше в абстрактный тест контроллеров.

VaolEr replied:
Ну либо в Util) Я вынесу уже после того, как добавлю эти изменения в dev.

VaolEr replied:
Сделал класс TestUtil. Вынес туда методы createTestUser & createTestAdmin.

VaolEr replied:
Вынес в AbstracControllerTest.

dimio commented on src/test/java/com/example/storehouse/web/categories/UserCategoriesControllerTests.java:
```diff
@@ -0,0 +1,111 @@
+        verifyNoInteractions(categoriesService);
+    }
+
+    //TODO решить как будет реализовано удаление данных при связанных талицах.
```
У нас же юзер не может удалять в любом случае, ему без разницы, как оно внутри реализовано. Запрос уже на уровне контроллера отвергается.

dimio commented on src/test/java/com/example/storehouse/web/categories/UserCategoriesControllerTests.java:
```diff
@@ -0,0 +1,111 @@
+        verifyNoInteractions(categoriesService);
+    }
+
+    //TODO вынести в файл TestData?
```
Как и с админом.

VaolEr replied:
Вынес в AbstracControllerTest.

VaolEr committed:
public static methods createTestUserAdmin() and createTestUserUser() created in AbstractControllerTest;
Fixed test classes where createTestUserAdmin() and createTestUserUser() were used;

VaolEr committed:
Add names to tests;
